extensions/Lily/MoreControl.js

```diff
@@ -0,0 +1,1214 @@
+        delete util.thread.newThread;
+        return returnValue ?? "";
+      } else {
+        util.thread.peekStackFrame().waitingReporter = true;
+        util.yield();
+      }
+    }
+
+    inlineReturn(args, util) {
+      util.thread.returnValue = args.VALUE;
+      util.stopThisScript();
+    }
+
+    elseIf(args) {
+      const condition1 = Cast.toBoolean(args.CONDITION1);
+      const condition2 = Cast.toBoolean(args.CONDITION2);
+      if (condition1) {
+        return 1;
+      } else if (condition2) {
+        return 2;
+      }
+    }
+
+    elseIfElse(args) {
+      const condition1 = Cast.toBoolean(args.CONDITION1);
+      const condition2 = Cast.toBoolean(args.CONDITION2);
+      if (condition1) {
+        return 1;
+      } else if (condition2) {
+        return 2;
+      } else {
+        return 3;
+      }
+    }
+
+    funnyBlock(args) {
+      for (const arg in args) {
+        if (args[arg]) {
+          return arg;
+        }
+      }
+    }
+
+    waitDuration(args, util) {
+      const type = Cast.toString(args.TYPE);
+      if (type == "frames") {
+        const duration = Math.round(Cast.toNumber(args.DURATION));
+        if (typeof util.stackFrame.loopCounter === "undefined") {
+          util.stackFrame.loopCounter = duration;
+        }
+        util.stackFrame.loopCounter--;
+        if (util.stackFrame.loopCounter >= 0) {
+          return true;
+        }
+      } else if (type == "seconds") {
+        if (util.stackTimerNeedsInit()) {
+          const duration = Math.max(0, 1000 * Cast.toNumber(args.DURATION));
+
+          util.startStackTimer(duration);
+          runtime.requestRedraw();
+          return true;
+        } else if (!util.stackTimerFinished()) {
+          return true;
+        }
+      }
+    }
+
+    waitDurationOrUntil(args, util) {
+      const type = Cast.toString(args.TYPE);
+      if (type == "frames") {
+        const duration = Math.round(Cast.toNumber(args.DURATION));
+        if (typeof util.stackFrame.loopCounter === "undefined") {
+          util.stackFrame.loopCounter = duration;
+        }
+        util.stackFrame.loopCounter--;
+        if (util.stackFrame.loopCounter >= 0 && !args.CONDITION) {
+          return true;
+        }
+      } else if (type == "seconds") {
+        if (util.stackTimerNeedsInit()) {
+          const duration = Math.max(0, 1000 * Cast.toNumber(args.DURATION));
+
+          util.startStackTimer(duration);
+          runtime.requestRedraw();
+          return true;
+        } else if (!util.stackTimerFinished() && !args.CONDITION) {
+          return true;
+        }
+      }
+    }
+
+    restart(args, util) {
+      runtime._restartThread(util.thread);
+    }
+
+    forArg(args, util) {
+      const param = "i";
+      const stackFrames = util.thread.stackFrames;
+      if (typeof stackFrames === "undefined") return 0;
+
+      const params = stackFrames[0].moreControlParams;
+      if (typeof params === "undefined") return 0;
```

Comment:
A problem I encountered while working on a project there should be a condition for json Objects in the JSON array. I don't know exactly how it is fetched from the stack but
there should be a check for the param being an object!
```js
if (typeof params[param] == "object") return JSON.stringify(params[param]);
```

before returning
```js
return params[param] ?? 0;
```

Reply:
Thank you! this will be fixed shortly